### PR TITLE
DIG-1409: Fix docker health checks

### PR DESCRIPTION
**Pull request by lilyyangyi301**
Screenshot 2024-01-30 at 4 10 27 PM

- Fixed all of the required docker containers' healthcheck, including opa, vault, minio, candig data portal, candig-ingest, federation, htsget, katsu, postgres. 
- For vault and opa, the healthcheck are switched to be done inside the runner container.
- For federation and opa, a healthcheck.py is added to resolve the authentication issue.

### CODE REVIEW

File: lib/opa/docker-compose.yml [hunk 1]
```diff
@@ -21,13 +21,23 @@ services:
...
       KEYCLOAK_CLIENT_ID: ${KEYCLOAK_CLIENT_ID}
+      KEYCLOAK_REALM_URL: ${KEYCLOAK_REALM_URL}
+      KEYCLOAK_URL: ${KEYCLOAK_PRIVATE_URL}
       OPA_SITE_ADMIN_KEY: ${OPA_SITE_ADMIN_KEY}
       OPA_URL: ${OPA_PRIVATE_URL}
-      KEYCLOAK_REALM_URL: ${KEYCLOAK_REALM_URL}
       VAULT_URL: ${VAULT_PRIVATE_URL}
       SERVICE_NAME: ${SERVICE_NAME}
+    healthcheck:
+      test: [ "CMD", "python", "/app/healthcheck.py"]
```
Comment (mshadbolt) [inline lib/opa/docker-compose.yml:38]: I think if there is a new script in opa you will need to commit it to the opa repo and make a PR there first (If I am understanding correctly)

File: lib/federation/docker-compose.yml [hunk 2]
```diff
@@ -32,3 +36,8 @@ services:
         SERVICE_NAME: ${SERVICE_NAME}
     extra_hosts:
         - "${CANDIG_DOMAIN}:${LOCAL_IP_ADDR}"
+    healthcheck:
+      test: ["CMD", "python", "/app/federation/candig_federation/healthcheck.py"]
```
Comment (mshadbolt) [inline lib/federation/docker-compose.yml]: I think the same for this healthcheck.py, you will need to commit it in federation and make a PR in that repo before updating in CanDIGv2

File: lib/opa/docker-compose.yml (hunk 1 above)
Comment (lilyyangyi301) [reply lib/opa/docker-compose.yml:38]: Added! Here is the [link](https://github.com/CanDIG/candig-opa/pull/42)!

File: lib/federation/docker-compose.yml (hunk 2 above)
Comment (lilyyangyi301) [reply lib/federation/docker-compose.yml]: Added! Here is the [link](https://github.com/CanDIG/federation_service/pull/48)!

**Review by mshadbolt — APPROVED** [review]
checked on my local stack and there is so much helth woo!

Screenshot 2024-02-06 at 10 43 41 AM

**Review by OrdiNeu — CHANGES_REQUESTED** [review]
Everything looks good! One last thing though -- I notice there's a lot of submodule updates that are here that aren't related to the health check -- candigv2-ingest, htsget_app, and a few others. Can you remove those from this PR (merge from develop for this repo, then only include the ones that actually changed)?

**Review by OrdiNeu — APPROVED** [review]
Fixed up the versions to develop's versions

**Pull request merged**